client-base/src/main/webapp/html/javascript/utils.js

```diff
@@ -1939,23 +1996,25 @@ function createSynonymTable(text) {
                 }
             }
 
+            var conceptKey = 'ConceptID:'+ keyword; 
+
             if (currentTarget.hasClass("addKeyword")) {
```

Comment:
Optional, this endless copy-pasting is ugly, produces too much code, which is hard to read and harder to maintain. Become fixated with reuse and compactness.  A better version here would be something like:

```javascript
className = currentTarget.getClass () // not sure this is the method
if ( [ "addKeyword", "addKeywordUndo", ... ].includes ( className ) )
  // As per https://stackoverflow.com/questions/359788
  window [ className ] ( conceptKey, currentTarget.attr ( "id" ), "keywords" );
else if ( [ "replaceKeyword", "replaceKeywordUndo", ... ].includes ( className ) )
   // The same, but has an additional parameter
    window [ className ] ( originalTermName, conceptKey, currentTarget.attr ( "id" ), "keywords" );
else
  // throw error?
```